### PR TITLE
Enable flight recorder for azure app service.

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import com.microsoft.azure.management.appservice.WebAppBase;
+import com.microsoft.azure.toolkit.lib.appservice.ProcessInfo;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.List;
+
+public abstract class FlightRecorderStarterBase {
+    protected WebAppBase appService;
+    protected PublishingProfile publishingProfile;
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java:35]: `publishingProfile` used in later PR

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/MyKuduClient.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import rx.Emitter;
+import rx.Emitter.BackpressureMode;
+import rx.Observable;
+import rx.functions.Action1;
+import rx.functions.Func1;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class MyKuduClient {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/MyKuduClient.java:50]: -> `FlyRecorderKuduClient` or `ExtendedKuduClient` ?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    abstract String getFinalJfrPath(String fileName);
+
+    protected String constructJcmdCommand(int pid, int timeInSeconds, String fileName) {
+        return String.format("jcmd %d JFR.start name=TimedRecording settings=profile duration=%ds filename=%s", pid,
+                             timeInSeconds, getFinalJfrPath(fileName));
+    }
+
+    public abstract CommandOutput startFlightRecorder(int pid, int timeInSeconds, String fileName) throws IOException;
+
+    public byte[] downloadJFRFile(String fileName) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java:55]: return Stream here? or add parameter `File dest` and write to File directly?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java (hunk 1 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java]: No, it should be removed

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java [hunk 4]
```diff
@@ -32,13 +31,11 @@
 
 public abstract class FlightRecorderStarterBase {
     protected WebAppBase appService;
-    protected PublishingProfile publishingProfile;
-    protected MyKuduClient client;
+    protected FlyRecorderKuduClient client;
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java:34]: `FlightRecorder` or `FlyRecorder`? I can see both.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java (hunk 3 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java:52]: Will handle the file download in later PR

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java (hunk 4 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/jfr/FlightRecorderStarterBase.java]: Done

**Review by Flanker32 — APPROVED** [review]

**Pull request merged**